lightning/src/util/events.rs

```diff
@@ -150,6 +150,25 @@ pub enum Event {
 		/// The outputs which you should store as spendable by you.
 		outputs: Vec<SpendableOutputDescriptor>,
 	},
+	/// This event is generated when a payment has been successfully forwarded through us and a
+	/// forwarding fee earned.
+	PaymentForwarded {
+		/// The fee, in milli-satoshis, which was earned as a result of the payment.
+		///
+		/// Note that if we force-closed the channel over which we forwarded an HTLC while the HTLC
+		/// was pending, the amount the next hop claimed may have been rounded down, causing a
+		/// higher fee than expected.
+		///
+		/// If the channel which sent us the payment has been force-closed, we will claim the funds
+		/// via an on-chain transaction. In that case we do not consider the on-chain transaction
+		/// fees involved in such a claim and this value will be `None`. It is possible duplicate
+		/// `PaymentForwarded` events are generated for the same payment iff `fee_earned_msat` is
+		/// `None`.
+		fee_earned_msat: Option<u64>,
+		/// If this is `true`, the forwarded HTLC will be claimed on-chain after the channel on
+		/// which we received the HTLC was force-closed.
```

Comment:
I think `claim_funds_internal()` in `process_pending_monitor_events` is called with from_onchain=true. Though this callsite at monitor events processing doesn't imply the backward channel has been force-closed, a check we only realize at the beginning of `claim_funds_from_hop`. 

So I think this comment is confusing...Shouldn't this say "we offered the HTLC was force-closed" to be correct ?

Reply:
Lol, these docs are just entirely wrong. I pushed a fixup commit.